### PR TITLE
Documentation improvements

### LINKED ISSUE #1323: Document the REST API
We need to document the REST API. The API had been automatically documented at some point in the past, but that stopped working and was removed (so that the docs could be built):

https://github.com/fedora-infra/bodhi/commit/b3f80b1b27e4b7ff2913d77fbae8bfb352d73e2e

We should either figure out how to get Cornice working again, find another automatic solution for documenting the REST API, or at least basically document it by hand as a last resort.

This is high priority as it will be needed for converting Bodhi to have multiple content types.

### PR DESCRIPTION
**Pull request by bowlofeggs**
This pull request has two commits. The first cleans up the man pages a bit and gets them building again. The second commit adds automatically generated REST API documentation using cornice.

Unfortunately, the upstream version of cornice is not currently capable of automatically generating these docs without two patches:

https://github.com/Cornices/cornice.ext.sphinx/pull/5
https://github.com/Cornices/cornice.ext.sphinx/pull/9

I have backported both of those patches for cornice-1.1.0 in Fedora 24-27, but unfortunately our Read the Docs builder can't use those packages.

fixes #1323

### CODE REVIEW

File: docs/conf.py
Comment (jeremycline) [inline docs/conf.py]: Do you still need this? I don't see a Python module/package for this.

File: docs/rest_api/index.rst
Comment (jeremycline) [inline docs/rest_api/index.rst:11]: You could just put all the REST API in this file (and maybe rename it to rest_api.rst). You'll still have a TOC to navigate to a section, but you don't have to click back and forth so much. I like it more in one file, but it's up to you.

**Review by jeremycline — APPROVED** [review]
Two minor things, neither of which have much impact. Otherwise looks good to me, :shipit:

File: docs/conf.py
Comment (bowlofeggs) [reply docs/conf.py]: Oh sorry, this was there from when I was wrestling with trying to figure out how to get RTD to work. I think I'm going to give up on RTD now that we have the docs hosted at https://bodhi.stg.fedoraproject.org/docs/

File: docs/rest_api/index.rst
Comment (bowlofeggs) [reply docs/rest_api/index.rst:11]: It was actually in one file when I started working on this and it was huge so I actually broke it up on purpose.

**Comment by bowlofeggs** [comment]
Lemme clean up this PR to remove my experimental RTD stuff.

File: docs/rest_api/index.rst
Comment (jeremycline) [reply docs/rest_api/index.rst:11]: Okay, works for me

**Comment by bowlofeggs** [comment]
OK I think this PR is now the way I want it to be if you would like to review it now. I plan on holding it for a while until the needed cornice build is stabilized in Fedora 24-25.

**Comment by jeremycline** [comment]
Looks good

**Comment by bowlofeggs** [comment]
I'm going to go ahead and merge this. We can use a buildroot override if we need to make a release, so the build isn't blocked on those cornice updates.

**Comment by bowlofeggs** [comment]
I forgot to force push my branch before merging it, so GitHub doesn't know I've merged a rebased version of this PR. It was merged as these commits:

https://github.com/fedora-infra/bodhi/commit/dc35bb060bee4ab8de1ae46331f9dcafd6e9f10e
https://github.com/fedora-infra/bodhi/commit/ca33459e9d60bc9d374fe938fe4ebb5f3eb8caf4

**Pull request closed without merging**
